docs/template-development.md

```diff
@@ -0,0 +1,86 @@
+---
+title: "Template Development"
+weight: 50
+---
+
+# Minimum template requirements
+
+Let's discuss the minimum components your template requires: `template` directory and `package.json` file.
+
+## 1. `template` directory
+
+The `template` directory stores files with the generated outputs. In other words, the Generator processes all the files stored in this directory.
+
+```js
+import { File, Text } from "@asyncapi/generator-react-sdk";
+
+export default function({ asyncapi, params, originalAsyncAPI }) {
+return (
+    <File name="asyncapi.md">
+    <Text>This is a markdown file for my application.</Text>
+    <Text>App name is: **{ asyncapi.info().title() }**</Text>
+    </File>
+);
+}
+```
+
+TThe above example of a`template/index.js` file shows the generation process result. The user also receives an `asyncapi.md` file with hardcoded and dynamic information extracted into the AsyncAPI file.
+
+Every template must depend on the [`@asyncapi/generator-react-sdk` package](https://github.com/asyncapi/generator-react-sdk), which contains a template file's basic components.
+
+## 2. `package.json` file
+
+Before the generation process runs, the Generator installs the template into its dependencies. A `package.json` is necessary to identify the template name.
+
+The following block shows an example `package.json` file that points to the [React Render Engine](react-render-engine.md) and necessary dependencies:
+
+```json
+{
+  "name": "myTemplate",
+  "generator": {
+    "renderer": "react"
+  },
+  "dependencies": {
+    "@asyncapi/generator-react-sdk": "^0.2.25"
+  }
+}
+```
+
+## Template Configuration
+
+The `package.json` file from the Generator must be configured so that it contains JSON objects that may have the following information. Some of the parameters used for template configuration are:
+
+|Name|Type|Description|
+|---|---|---|
+|`renderer`| String | Its value can be either `react` or `nunjucks` (default).
+|`supportedProtocols`| [String] | A list with all the protocols this template supports.
+|`parameters`| Object[String, Object] | An object with all the parameters that can be passed when generating the template. When using the command line, it's done by indicating `--param name=value` or `-p name=value`.
+|`parameters[param].description`| String | A user-friendly description about the parameter.
+|`parameters[param].default`| Any | Default value of the parameter if not specified. Shouldn't be used for mandatory `required=true` parameters.
+|`parameters[param].required`| Boolean | Whether the parameter is required or not.
+
+These are predefined configurations which help generator achieve specific set of tasks throughout the generation process. The `generator` property from 'package.json' contains all the configuration information. To learn more about template configuration and various supported parameters. Read more about the [configuration file](configuration-file.md)
```

Comment:
```suggestion
These predefined configurations help the generator achieve specific tasks throughout the generation process. The `generator` property from the 'package.json' contains all the configuration information. To learn more about template configuration and various supported parameters, read the [configuration file doc](configuration-file.md)
```

Reply:
hey there! yes there is a doc for configuration file currently planned